src/jitlayers.cpp

```diff
@@ -129,12 +130,16 @@ void addOptimizationPasses(PassManager *PM)
     if (jl_options.opt_level == 0) {
         PM->add(createCFGSimplificationPass()); // Clean up disgusting code
         PM->add(createMemCpyOptPass()); // Remove memcpy / form memset
-        PM->add(createLowerPTLSPass(imaging_mode));
 #if JL_LLVM_VERSION >= 40000
         PM->add(createAlwaysInlinerLegacyPass()); // Respect always_inline
 #else
         PM->add(createAlwaysInlinerPass()); // Respect always_inline
 #endif
```

Comment:
Yes. It only looks at our address spaces. If external code uses our address spaces, it needs to follow our invariants.